crates/turborepo-lib/src/daemon/server.rs

```diff
@@ -548,6 +557,55 @@ impl proto::turbod_server::Turbod for TurboGrpcServiceInner {
                 }
             })
     }
+
+    type PackageChangesStream = ReceiverStream<Result<proto::PackageChangeEvent, tonic::Status>>;
+
+    async fn package_changes(
+        &self,
+        _request: tonic::Request<proto::PackageChangesRequest>,
+    ) -> Result<tonic::Response<Self::PackageChangesStream>, tonic::Status> {
+        let mut package_changes_rx = self.file_watching.package_changes_watcher.package_changes();
+        let (tx, rx) = mpsc::channel(1);
+
+        tx.send(Ok(proto::PackageChangeEvent {
+            event: Some(proto::package_change_event::Event::RediscoverPackages(
+                proto::RediscoverPackages {},
+            )),
+        }))
+        .await
+        .map_err(|e| tonic::Status::unavailable(format!("{}", e)))?;
+
+        tokio::spawn(async move {
+            loop {
+                if let Err(err) = package_changes_rx.changed().await {
+                    error!("package changes stream closed: {}", err);
```

Comment:
Is there a way to send an error, or close the stream with an error? Or at least close the stream?

Reply:
Since this breaks the loop, it should drop the `tx` which in turn closes the channel. But we can send errors instead I suppose.